File: 1616-minimum-difference-between-largest-and-smallest-value-in-three-moves/minimum-difference-between-largest-and-smallest-value-in-three-moves.py
class Solution:
    def minDifference(self, nums: List[int]) -> int:
        if len(nums)<=4:
            return 0
        nums.sort()
        minDiff=float('inf')
        for i in range(4):
            L,R=i,len(nums)-1 -(3-i)
            diff=nums[R]-nums[L]
            minDiff=min(minDiff,diff)
        return minDiff
            

File: 1616-minimum-difference-between-largest-and-smallest-value-in-three-moves/minimum-difference-between-largest-and-smallest-value-in-three-moves.cpp
class Solution {
public:
    int minDifference(vector<int>& nums) {
        if(nums.size()<=4)
            return 0;
        sort(nums.begin(),nums.end());
        int minDiff=INT_MAX;
        for(int i=0;i<4;i++){
            int L=i,R=nums.size()-1-(3-i);
            minDiff=min(minDiff,nums[R]-nums[L]);
        }
        return minDiff;
    }
};